src/main/scala/org/biodatageeks/sequila/rangejoins/methods/IntervalTree/IntervalHolderChromosome.scala

```diff
@@ -3,7 +3,7 @@ package org.biodatageeks.sequila.rangejoins.methods.IntervalTree
 import org.biodatageeks.sequila.rangejoins.IntervalTree.Interval
 import org.biodatageeks.sequila.rangejoins.methods.base.BaseIntervalHolder
 
-class IntervalHolderChromosome[T](allRegions: Array[(String,Interval[Int],T)], intervalHolderClassName:String) extends Serializable {
+class IntervalHolderChromosome[T](allRegions: Array[(String,Interval[Int],T)], intervalHolderClassName:String, domains: Option[Int] = None) extends Serializable {
```

Comment:
I would suggest passing a generic Conf-type object which is a key/value instead of passing it explicitly.